File: server/src/Core/Entities/Game.cs
namespace ChessProject.Core.Entities;

public class Game
{
    public Guid Id { get; set; }
    public string? WhitePlayerId { get; set; }
    public string? BlackPlayerId { get; set; }

    public string FEN { get; set; } = "rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1";

    public string MoveHistory { get; set; } = "";

    public string Status { get; set; } = "WAITING"; // WAITING, PLAYING, FINISHED

    public string? WinnerId { get; set; }

    public long WhiteTimeRemainingMs { get; set; } = 10 * 60 * 1000;
    public long BlackTimeRemainingMs { get; set; } = 10 * 60 * 1000;

    public int TimeLimitMinutes { get; set; } = 10; // Mặc định 10 phút
    public int IncrementSeconds { get; set; } = 0;  // Mặc định 0 giây

    public DateTime? LastMoveAt { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime? FinishedAt { get; set; }
}

File: client/src/types/game.ts
export interface Game {
  id: string;
  whitePlayerId: string;
  blackPlayerId: string;
  fen: string;
  createdAt: string;
  status: "WAITING" | "PLAYING" | "FINISHED";
  whiteUsername?: string | null;
  blackUsername?: string | null;

  whiteTimeRemainingMs: number;
  blackTimeRemainingMs: number;
<<<<<<< HEAD
  incrementMs?: number;
=======

  whiteUsername?: string;
  blackUsername?: string;

>>>>>>> 3efbb6bcef957fb04fcd970e30cda3a5f228393d
  lastMoveAt: string | null;

  winnerId: string | null;

  moveHistory: string;
}

export interface PaginatedResult<T> {
  items: T[];
  totalCount: number;
  pageIndex: number;
  pageSize: number;
  totalPages: number;
}
